ui/StatusQ/src/StatusQ/Components/StatusChatList.qml

```diff
@@ -40,238 +43,164 @@ Item {
+                horizontalPadding:0
+                verticalPadding: 0
+                icon.width: 0
+                icon.height: 0
+                spacing: 0
+                topInset: 0
+                bottomInset: 0
+                customizable: true
+                onClicked: {
+                    if (model.isCategory) {
+                        statusChatListCategoryItem.clicked(mouse);
+                    } else {
+                        statusChatListItem.clicked(mouse);
                     }
                 }
-            }
-            
-            Component {
-                id: channelItemComponent
-                QC.Control {
-                    id: draggable
-                    objectName: model.name
-                    width: root.width
-                    height: model.categoryOpened ? statusChatListItem.height + 4 /*spacing between non-collapsed items*/ : 0
-                    visible: height
-                    verticalPadding: 2
-
-                    property alias chatListItem: statusChatListItem
-
-                    contentItem: MouseArea {
-                        id: dragSensor
-
-                        anchors.fill: parent
-                        cursorShape: active ? Qt.ClosedHandCursor : Qt.PointingHandCursor
-                        hoverEnabled: true
-                        enabled: root.draggableItems
-
-                        property bool active: false
-                        property real startY: 0
-                        property real startX: 0
-
-                        drag.target: draggedListItemLoader.item
-                        drag.filterChildren: true
-
-                        onPressed: {
-                            startY = mouseY
-                            startX = mouseX
+                actions: [
+                    StatusChatListCategoryItem {
+                        id: statusChatListCategoryItem
+                        objectName: "categoryItem"
+                        visible: model.isCategory
+                        function setupPopup() {
+                            categoryPopupMenuSlot.item.categoryItem = model
                         }
-                        onPressAndHold: active = true
-                        onReleased: {
-                            if (active && d.destinationPosition !== -1 && statusChatListItem.originalOrder !== d.destinationPosition) {
-                                root.chatItemReordered(statusChatListItem.chatId, statusChatListItem.originalOrder, d.destinationPosition)
+                        Connections {
+                            enabled: categoryPopupMenuSlot.active && statusChatListCategoryItem.highlighted
+                            target: categoryPopupMenuSlot.item
+                            function onClosed() {
+                                statusChatListCategoryItem.highlighted = false
+                                statusChatListCategoryItem.menuButton.highlighted = false
```

Comment:
Couldn't these be bindings? And not:
- part of the delegate
- imperative JS code

Reply:
this was existing code, and it's a handling the popup's closing, how/where would you suggest moving it otherwise?

Reply:
nvm for now, just keep it :)